registry/txt.go

```diff
@@ -34,6 +35,26 @@ const (
 	providerSpecificForceUpdate = "txt/force-update"
 )
 
+type TXTFormat string
+
+const (
+	TXTFormatTransition   TXTFormat = "transition"
+	TXTFormatOnlyMetadata TXTFormat = "only-metadata"
```

Comment:
This looks like an enum and we should use this:
```
type TXTFormat int

const (
	TXTFormatTransition TXTFormat = iota + 1
	TXTFormatOnlyMetadata
)

func (t TXTFormat) String() string {
	switch t {
	case TXTFormatTransition:
		return "transition"
	case TXTFormatOnlyMetadata:
                return "only-metadata"
	}
        return "unknown"
}
```

Reply:
I decided to simplify and stick to strings, in alignment with the code style of the project (as in [here](https://github.com/kubernetes-sigs/external-dns/blob/master/endpoint/endpoint.go#L30))

Validation of these values is done as per user input: